core/src/main/java/org/apache/iceberg/IncrementalDataTableScan.java

```diff
@@ -46,6 +46,14 @@ public TableScan asOfTime(long timestampMillis) {
             timestampMillis, context().fromSnapshotId(), context().toSnapshotId()));
   }
 
+  @Override
+  public TableScan useRef(String ref) {
+    throw new UnsupportedOperationException(
```

Comment:
This seems fine for now, but we may have a case where we want incremental to be able to specify a branch. It doesn't matter currently because the snapshot IDs are always explicit.